### ISSUE TITLE
[feature-request] [help-request] Export functionality

### ISSUE THREAD

**Issue by philipptempel** [getgrav/grav-plugin-data-manager#18]
For my department, I deployed grav as a simple one-page event CMS that can host multiple events on different hosts. Setting everything up (modular pages, admin plugin and other plugins) was a breeze. However, now it was brought to my attention that the registration form submissions (stored and displayed using the data-manager plugin) should also be exported as e.g., CSV. In general, I would say that my skills for developing themes and plugins for grav are quite good, however, for the better part of the last two weeks I could not figure out a way to implement an export feature in the plugin.

I was wanting to create two export functionalities, one on a per-type level and one on a per-item level (speaking in data-manger jargon). Doing all the processing and creating the download isn't a problem (thanks to looking into other plugins' code), yet the problem arises from integrating the form into the existing template files - or making a standard HTML form in the template files work.

The main export functionality is added to the ```onFormProcessed``` event callback, to which I have the plugin register. Within the callback, I check for all the right things and create the correct files, but this function never gets triggered.

Are there any plans on integrating an export functionality into the plugin? Or would there be any good tutorials on this topic available around the grav pages?

**Comment by garytube** [comment]
@philipptempel  i'm also looking into a export function, but i'm stuck. have you any code to share?

**Comment by philipptempel** [comment]
I had the code on a private repo that I cannot share in its entirety. That's why I just forked the original `grav-plugin-data-manager` project and created a [feature branch](https://github.com/philipptempel/grav-plugin-data-manager/tree/feature/export-data-into-csv) which contains my code for export.
The export functionality was created shortly after I opened this issue, however, it's sort of more a "just make it work" rather than implemented in a proper grav'ish way.

You will need to run `composer update` once you've cloned the repository to get in all the dependencies for CSV export.
